TestModels/SharedMakefile.mk

```diff
@@ -162,15 +224,14 @@ transpile_test:
 		-functionSyntax:3 \
 		-useRuntimeLib \
 		-out $(OUT) \
-		-library:$(PROJECT_ROOT)/dafny-dependencies/StandardLibrary/src/Index.dfy \
+		$(if $(strip $(STD_LIBRARY)) , -library:$(PROJECT_ROOT)/$(STD_LIBRARY)/src/Index.dfy, ) \
```

Comment:
Sorry, what is this `if strip` logic?

Reply:
The problem this is solving is that the StandardLibrary is automatically considered a dependency of all libraries, but it can't be when building the StandardLibrary itself. 

The TestModels folder's solution is to override lots of targets in `StandardLibrary/Makefile` to do nothing. The MPL, ESDK and DB ESDK use this solution instead: only consider the StandardLibrary a dependency if `STD_LIBRARY` is empty (after stripping whitespace, which I gather is easy to introduce in make variables if you're not careful).

I opted to adopt the latter solution since it aligned a little better with reality and would cause less churn when adopting the SmithyDafnyMakefile.mk in the MPL, ESDK and DB ESDK.

Reply:
I suggest a comment somewhere.